lib/std/os/windows/user32.zig

```diff
@@ -28,7 +28,7 @@ const POINT = windows.POINT;
 const HCURSOR = windows.HCURSOR;
 const HBRUSH = windows.HBRUSH;
 
-fn selectSymbol(comptime function_static: anytype, function_dynamic: @TypeOf(function_static), comptime os: std.Target.Os.WindowsVersion) @TypeOf(function_static) {
+fn selectSymbol(comptime function_static: anytype, function_dynamic: *const @TypeOf(function_static), comptime os: std.Target.Os.WindowsVersion) *const @TypeOf(function_static) {
```

Comment:
The return type should remain what it was before when `sym_ok == true`. Only when `sym_ok == null` should this return a function pointer.

Reply:
Oh, i see. Thanks. I got lost a little.

```zig
fn selectSymbol(
    comptime function_static: anytype,
    function_dynamic: *const @TypeOf(function_static),
    comptime os: std.Target.Os.WindowsVersion,
) if (builtin.os.isAtLeast(.windows, os) == null) @TypeOf(function_static) else @TypeOf(function_dynamic) {
    comptime {
        const sym_ok = builtin.os.isAtLeast(.windows, os);
        if (sym_ok == true) return function_static;
        if (sym_ok == null) return function_dynamic;
        if (sym_ok == false) @compileError("Target OS range does not support function, at least " ++ @tagName(os) ++ " is required");
    }
}
```

I don't understand this code after all. Return type if dynamic must be @TypeOf(function_dynamic) or else @TypeOf(function_static). But it is working if types are reversed. Do you think my changes are correct?

Reply:
If i reverse types i got next error:
```
.local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:1589:48: error: unable to resolve comptime value
    const function = selectSymbol(MessageBoxW, pfnMessageBoxW, .win2k);
                                               ^~~~~~~~~~~~~~
.local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:1589:48: note: argument to function being called at comptime must be comptime-known
.local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:35:3: note: expression is evaluated at comptime because the generic function was instantiated with a comptime-only return type
) if (builtin.os.isAtLeast(.windows, os) == true) @TypeOf(function_static) else @TypeOf(function_dynamic) {
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    messageBoxW: .local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:1589:22
    main: main.zig:4:34
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

Reply:
The return type should be:
```zig
if (builtin.os.isAtLeast(.windows, os) == null) *const @TypeOf(function_static) else @TypeOf(function_static)
```

This returns a function pointer when `sym_ok` equals to `null` and returns a function body when not.

Reply:
Then this code is not compiling:
```zig
_ = try std.os.windows.user32.messageBoxW(null, std.unicode.utf8ToUtf16LeStringLiteral("hello"), std.unicode.utf8ToUtf16LeStringLiteral("world"), 0);
```

```
.local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:1589:48: error: unable to resolve comptime value
    const function = selectSymbol(MessageBoxW, pfnMessageBoxW, .win2k);
                                               ^~~~~~~~~~~~~~
.local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:1589:48: note: argument to function being called at comptime must be comptime-known
.local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:35:3: note: expression is evaluated at comptime because the generic function was instantiated with a comptime-only return type
) if (builtin.os.isAtLeast(.windows, os) == null) *const @TypeOf(function_static) else @TypeOf(function_static) {
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    messageBoxW: .local/zig/ver/zig-linux-x86_64-0.11.0-dev.1268+0e66df209/lib/std/os/windows/user32.zig:1589:22
    main: main.zig:4:34
    remaining reference traces hidden; use '-freference-trace' to see all reference traces

```

Reply:
Also, i think we should drop this code. It is checking for windows 2000. Zig doesn't work on windows 7.